lib/client/backend_client.dart

```diff
@@ -11,8 +11,9 @@ class BackendClient {
   BackendClient({http.Client? httpClient})
       : httpClient = httpClient ?? http.Client();
 
-  Future<dynamic> getData({required String uri}) async {
-    var url = Uri.http(localhost, uri);
+  Future<dynamic> getData(
+      {required String uri, Map<String, dynamic>? queryParams}) async {
+    var url = Uri.http(localhost, uri, queryParams);
```

Comment:
I recommend changing this to:

```suggestion
    String url = Uri.http(localhost, uri, queryParams);
```

It already has type-safety because Dart knows the type, but off the top of my head, I don't know what `Uri.http()` returns.

Reply:
got it